driver/src/main/scala/com/stratio/sparkta/driver/dto/AggregationPoliciesDto.scala

```diff
@@ -93,20 +97,49 @@ object AggregationPoliciesValidator {
     (isValid, msg)
   }
 
-  def validateRollups(aggregationPoliciesDto: AggregationPoliciesDto): (Boolean, String) = {
+  def validateCubes(aggregationPoliciesDto: AggregationPoliciesDto): (Boolean, String) = {
 
+    val hasCubesNames = aggregationPoliciesDto.cubes.filter(c => c.name == null || c.name.isEmpty).isEmpty
     val dimensionNames = aggregationPoliciesDto.dimensions.map(_.name)
-
-    val rollupNames = aggregationPoliciesDto.rollups
+    val cubeDimension = aggregationPoliciesDto.cubes
       .flatMap(x => Option(x))
-      .flatMap(x => Option(x.dimensionAndPrecision))
+      .flatMap(x => Option(x.dimensions))
       .flatten
-      .map(_.dimensionName)
+      .map(_.dimension)
+
+    if(!hasCubesNames) (false, MessageCubeName)
+    else checkCubeParameter(cubeDimension, dimensionNames, "dimensions") match {
+      case resultDimension if resultDimension._1 => resultDimension
+      case _ => {
+        val operatorNames = aggregationPoliciesDto.operators.map(_.name)
+        val cubeOperators = aggregationPoliciesDto.cubes
+          .flatMap(x => Option(x))
+          .flatMap(x => Option(x.operators))
+          .flatten
+
+        checkCubeParameter(cubeOperators, operatorNames, "operators") match {
+          case resultOperator if resultOperator._1 => resultOperator
+          case _ => {
+            val outputNames = aggregationPoliciesDto.outputs.map(_.name)
+            val cubeOutputs = aggregationPoliciesDto.cubes
+              .flatMap(x => Option(x))
+              .flatMap(x => Option(x.outputs))
+              .flatten
+
+            checkCubeParameter(cubeOutputs, outputNames, "outputs")
+          }
+        }
+      }
+    }
+  }
+
+  private def checkCubeParameter(cubeNames: Seq[String], parameterNames: Seq[String], label: String):
+  (Boolean, String) = {
+    val parameterNotIn = cubeNames.filter(!parameterNames.contains(_))
+    val isParameterIn = parameterNotIn.isEmpty
+    val isCubeInMsg =
+      if (!isParameterIn) s"All $label cubes should be declared in $label block\n" else ""
```

Comment:
This message is wierd. Set something like "All references to $label in cubes should be declared in $label block"